server/data/data-library/src/main/java/org/apache/james/domainlist/lib/DomainListImpl.java

```diff
@@ -62,25 +59,32 @@
         DetectedIp
     }
 
-    public static final String CONFIGURE_AUTODETECT = "autodetect";
-    public static final String CONFIGURE_AUTODETECT_IP = "autodetectIP";
-    public static final String CONFIGURE_DEFAULT_DOMAIN = "defaultDomain";
-    public static final String CONFIGURE_DOMAIN_NAMES = "domainnames.domainname";
-    public static final String ENV_DOMAIN = "DOMAIN";
+    static final String ENV_DOMAIN = "DOMAIN";
 
     private final DNSService dns;
+    protected final T domainListDAO;
     private final EnvDetector envDetector;
     private boolean autoDetect = true;
     private boolean autoDetectIP = true;
     private Domain defaultDomain;
 
-    public AbstractDomainList(DNSService dns, EnvDetector envDetector) {
+    @Inject
+    DomainListImpl(DNSService dns, T domainListDAO, EnvDetector envDetector) {
         this.dns = dns;
+        this.domainListDAO = domainListDAO;
         this.envDetector = envDetector;
     }
 
-    public AbstractDomainList(DNSService dns) {
-        this(dns, new EnvDetector());
+    public DomainListImpl(DNSService dns, T domainListDAO) {
+        this(dns, domainListDAO, new EnvDetector());
+    }
+
+    @Override
+    public void addDomain(Domain domain) throws DomainListException {
```

Comment:
this method is a good example of why this kind of abstraction doesn't bring much value: contains followed by add is almost never working because of race conditions.